Upstream decided to "Stop using -fzero-call-used-regs=all
... since it seems to be problematic with several different versions of
clang."

https://github.com/openssh/openssh-portable/commit/ff220d4010717f7bfbbc02a2400666fb9d24f250

But we don't build openssh with clang and also have issues with
zero-call-used-regs=used on ppc64le:

  sorry, unimplemented: argument 'used' is not supported for '-fzero-call-used-regs' on this target

--- a/configure.ac
+++ b/configure.ac
@@ -236,7 +236,7 @@ if test "$GCC" = "yes" || test "$GCC" =
 	case "$CLANG_VER" in
 	apple-15*) OSSH_CHECK_CFLAG_LINK([-fzero-call-used-regs=used]) ;;
 	17*)	;;
-	*)	OSSH_CHECK_CFLAG_LINK([-fzero-call-used-regs=used]) ;;
+	*)	OSSH_CHECK_CFLAG_LINK([-fzero-call-used-regs=all]) ;;
 	esac
 	OSSH_CHECK_CFLAG_COMPILE([-ftrivial-auto-var-init=zero])
     fi
